### PR TITLE
Write a test of download module

### PR DESCRIPTION
**Pull request by unnonouno**
I made a test of download.

### CODE REVIEW

**Comment by mitmul** [comment]
LGTM

File: chainer/dataset/download.py [hunk 1]
```diff
@@ -139,6 +139,12 @@ def cache_or_load_file(path, creator, loader):
     temp_dir = tempfile.mkdtemp()
     temp_path = os.path.join(temp_dir, file_name)
 
+    try:
+        os.makedirs(_dataset_root)
+    except OSError:
+        if not os.path.exists(_dataset_root):
+            raise RuntimeError('cannot create dataset directory')
```
Comment (gwtnb) [inline chainer/dataset/download.py]: When `_dataset_root` exists but is not a directory, an exception is lost unexpectedly.

**Comment by unnonouno** [comment]
I updated it.

**Comment by mitmul** [comment]
The test of Appveyor failed. Could you check it again? @unnonouno

**Comment by okuta** [comment]
LGTM

**Pull request merged**